### ISSUE TITLE
Large files are missing faces/textures when imported in URP however works fine in Standard Rendering Pipeline

### ISSUE THREAD

**Issue by subhradeepc** [atteneder/glTFast#243]
Large embedded/textured gltf files are missing faces when imported at designtime/runtime in Universal Rendering Pipeline , but works properly with the library when the project is in standard pipeline.

File:
https://drive.google.com/file/d/19LHoLsB0Ek1wfhyUmQpnrHQ3sNz02w94/view?usp=sharing

1. Create and empty project in URP 
2. Install the library
3. Download the above gltf file and import it
4. Notice that faces are missing.

Unity Editor version 2021.1.14 Mac
 - Render Pipeline and version : Universal Render Pipeline

Standard Pipeline:-
![Screenshot 2021-09-09 at 7 54 56 AM](https://user-images.githubusercontent.com/33677582/132612093-dfaf3232-5491-4959-8202-ff3dfba0e09b.png)

Universal Pipeline:-
![Screenshot 2021-09-09 at 7 52 09 AM](https://user-images.githubusercontent.com/33677582/132612171-af79c293-612a-476f-8ff3-66f20e511fce.png)

**Comment by subhradeepc** [comment]
@atteneder Would really appreciate if you can help resolve this or point out where im going wrong, thanks in advance!

**Comment by subhradeepc** [comment]
Also noticed that, since in URP the materials point out to a different set of shaders (which is obvious i believe), so i tried switching the material's shaders, and it looks like assigning the "gltf-unlit-opaque-double" seems to work, however the unlit shader is obviously not a suitable one. 


![Screenshot 2021-09-09 at 8 55 52 AM](https://user-images.githubusercontent.com/33677582/132617493-6c47ddeb-2ba5-442d-b521-a1aefdb235a5.png)

**Comment by atteneder** [comment]
Thanks for the detailed report.

I'll look into it today/tomorrow.

**Comment by atteneder** [comment]
After a first glance:

It seems like the URP pipeline does not respect that your materials should be rendered double-sided (so back-face culling should be off)

Here's your model in Blender with back-face culling on. As you can see, it looks broken as well:
<img width="798" alt="image" src="https://user-images.githubusercontent.com/2593014/132649917-87a4d07a-447d-47fa-b9ee-7ba1e9351e16.png">

Apart from fixing this, I'd recommend the following:

- Make your normals consistent, so that faces of similar orientation are not flipped
- Set 'doubleSided' to false on all your materials. This will speed up rendering a lot
- Finally: Convert your glTFs to binary versions, preferably compressed (so no embedded buffers).

**Comment by atteneder** [comment]
TL;DR: This will not be fixed in near future. Please fix your model instead.

Alright, here's what I found out:

The materials glTFast generates on URP/HDRP are based on the Lit shader, which does have a double-sided mode (Render Face: Both). Though it seems this does not add another (costly) render pass that inverts normals/tangents properly to make sure the light response is calculated correctly. Hence the incorrect result.

Your materials are kinda broken / misconfigured as well. If you turn down the metallic factor to zero and reduce the smoothness, you'll see the basecolor texture re-appear (but the shading still is wrong).

So again, here's what you can do:

- Make your normals consistent, so that faces of similar orientation are not flipped
- Set 'doubleSided' to false on all your materials. This will speed up rendering a lot. If you absolutely need this, a workaround is to duplicate the geometry and flip the duplicated faces.

**Comment by wangyanmeego** [comment]
I have the same problem, it works well in editor, but when I build on PC. the texture did not appear. I use URP pipeline and I also add all URP shader in graphic setting.
In the Editor：
![Snipaste_2021-09-14_21-02-09](https://user-images.githubusercontent.com/13783955/133262426-df8d0c4a-8a31-49a5-af55-9d6ae565192d.png)

Build on windows：
![Snipaste_2021-09-14_20-57-34](https://user-images.githubusercontent.com/13783955/133261968-e7ee74ca-aad3-4ee1-af34-5524408333a2.png)

**Comment by atteneder** [comment]
> I have the same problem, it works well in editor, but when I build on PC. the texture did not appear. I use URP pipeline and I also add all URP shader in graphic setting.

It's hard to tell the problem without more information. Can you provide a file that reprodices the error? Please open a new, dedicated issue and fill out the form with more details.
